json/store.go

```diff
@@ -37,6 +37,11 @@ func (store BinaryStore) MarshalWithMetadata(tree sops.TreeBranch, metadata sops
 	return store.store.MarshalWithMetadata(tree, metadata)
 }
 
+// MarshalValue is unusable for BinaryStore
+func (store BinaryStore) MarshalValue(v interface{}) ([]byte, error) {
+	panic("BinaryStore can not marshal a single value")
```

Comment:
I'm concerned that the stack trace returned by a panic will prevent users from understanding they made a mistake in their command line flags, whereas an explicit error help them.

I think the error message should be something like `error: binary files are not structured and extracting a single value is not possible.`